src/init/init.c

```diff
@@ -229,6 +229,18 @@ static log_timestamp_t default_get_timestamp(void)
 }
 #endif
 
+#if CONFIG_IPC_MAJOR_4
+static void init_fw_reg(struct sof *sof)
+{
+	/* Set current abi version of the IPC4 FwRegisters layout */
+	size_t ipc4_abi_ver_offset = offsetof(struct ipc4_fw_registers, abi_ver);
+
+	mailbox_sw_reg_write(ipc4_abi_ver_offset, IPC4_FW_REGS_ABI_VER);
+
+	k_spinlock_init(&sof->fw_reg_lock);
```

Comment:
why not just call this directly in `primary_core_init()`? It doesn't seem to fit the name of this function - it doesn't initialise any registers.

Reply:
sure, let's do it in primary_core_init